prow/gcs/reporter/reporter.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reporter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/testgrid/metadata"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/googleapi"
+	"k8s.io/test-infra/prow/errorutil"
+
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gcsupload"
+	"k8s.io/test-infra/prow/pod-utils/downwardapi"
+)
+
+const reporterName = "gcsreporter"
+
+type gcsReporter struct {
+	cfg     config.Getter
+	dryRun  bool
+	logger  *logrus.Entry
+	storage *storage.Client
+}
+
+func (gr *gcsReporter) Report(pj *prowv1.ProwJob) ([]*prowv1.ProwJob, error) {
+	ctx := context.Background() // TODO: get this from somewhere better?
+
+	_, _, err := gr.getJobDestination(pj)
+	if err != nil {
+		gr.logger.Infof("Not uploading %q (%s#%s) because we couldn't find a destination.", pj.Name, pj.Spec.Job, pj.Status.BuildID)
```

Comment:
I think it makes sense here - we _should_ report, but we find we are unable to upon making the attempt.

(In practice I do not think this will ever occur on any remotely reasonable prow configuration.)

I don't really have strong preferences, though.

Reply:
imho no. this error is during reporting and I would think unexpected for normal operation?

Reply:
I'm fine leaving this and `ShouldReport` as-is then